plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -206,14 +229,28 @@ public void modifyText(ModifyEvent event) {
     String[] filterExtensions = new String[] {"*.json"}; //$NON-NLS-1$
     browse.addSelectionListener(new FileFieldSetter(serviceKeyInput, filterExtensions));
 
+    serviceKeyDecoration = new ControlDecoration(serviceKeyInput, SWT.LEAD | SWT.TOP);
+    serviceKeyDecoration.hide();
+
+    createServiceKey = new Button(composite, SWT.NONE);
+    createServiceKey.setText(Messages.getString("create.new.service.key")); //$NON-NLS-1$
+    createServiceKey.addSelectionListener(new SelectionAdapter() {
+      @Override
+      public void widgetSelected(SelectionEvent event) {
+        createServiceAccountKey(getServiceAccountKeyPath());
```

Comment:
Should wrap in a `getLaunchConfigurationDialog().run()`.

Reply:
Good idea. I think a `BusyIndicator` will work fine. This is a one API call, and we have very short timeouts set.